pkgs/build-support/node/build-npm-package/hooks/npm-install-hook.sh

```diff
@@ -0,0 +1,30 @@
+npmInstallHook() {
+    echo "Executing npmInstallHook"
+
+    runHook preInstall
+
+    # `npm pack` writes to cache
```

Comment:
```suggestion
    # `npm pack` writes the cache to $out, while it is only needed at build-time
```

We discussed this earlier in private, but I figured I'd leave it as a review comment here so others can get their opinions in as well.